File: app/src/main/java/org/hesab/app/Transaction.java
package org.hesab.app;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "transactions")
public class Transaction {

    @PrimaryKey(autoGenerate = true)
    public int id;

    public String date;
    public long amount;
    public String category;
    public String description;
    public boolean isIncome;
    public String bankName;
    public int orderIndex;

    public Transaction(String date, long amount, String category, String description,
                       boolean isIncome, String bankName, int orderIndex) {
        this.date = date;
        this.amount = amount;
        this.category = category;
        this.description = description;
        this.isIncome = isIncome;
        this.bankName = bankName;
        this.orderIndex = orderIndex;
    }
}

File: app/src/main/java/org/hesab/app/Transaction.kt
package org.hesab.app

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,

    val date: Long = System.currentTimeMillis(),   // تاریخ به صورت timestamp
    val amount: Long,                              // مبلغ
    val category: String,                          // بابت
    val note: String? = null,                      // توضیحات
    val isIncome: Boolean = false,                 // درآمد یا هزینه
    val orderIndex: Int = 0                        // ترتیب نمایش
)
